File: CatenaccioStore/client/src/app/models/Order.ts
import { Address } from "./Address"
import { DeliveryMethod } from "./DeliveryMethod"

export interface OrderToCreate{
    basketId: string;
    deliveryMethodId: number;
    shipToAddress: Address
}

export interface Order {
    buyerEmail: string
    orderDate: string
    shipToAddress: Address
    deliveryMethod: string //Maybe DeliveryMethod?
    orderItems: OrderItem[]
    subtotal: number
    total: number
    orderStatus: number
    paymentIntentId: any
    id: number
  }
  
  export interface OrderItem {
    itemOrdered: ItemOrdered
    price: number
    quantity: number
    id: number
  }
  
  export interface ItemOrdered {
    productItemId: number
    productName: string
    pictureUrl: string
  }
  

File: CatenaccioStore/CatenaccioStore.Core/Entities/Orders/Order.cs
﻿namespace CatenaccioStore.Core.Entities.Orders
{
    public class Order : BaseEntity
    {
        public Order()
        {
        }
        public Order(IReadOnlyList<OrderItem> orderItems,string buyerEmail, string? paymentIntentId, Address shipToAddress, DeliveryMethod deliveryMethod, decimal subtotal)
        {
            BuyerEmail = buyerEmail;
            ShipToAddress = shipToAddress;
            DeliveryMethod = deliveryMethod;
            OrderItems = orderItems;
            Subtotal = subtotal;
            PaymentIntentId = paymentIntentId;
        }

        public string BuyerEmail { get; set; }
        public DateTime OrderDate { get; set; } = DateTime.UtcNow;
        public Address ShipToAddress { get; set; }
        public DeliveryMethod DeliveryMethod { get; set; }
        public IReadOnlyList<OrderItem> OrderItems { get; set; }
        public decimal Subtotal { get; set; }
        public OrderStatus OrderStatus { get; set; } = OrderStatus.Pending;
        public string? PaymentIntentId { get; set; }
        public decimal GetTotal()
        {
            return Subtotal + DeliveryMethod.Price;
        }

    }
}
